src/blocks/map/components/edit.js

```diff
@@ -47,6 +47,7 @@ class Edit extends Component {
 		this.state = {
 			apiKey: '',
 			address: this.props.attributes.address,
+			addressText: '',
```

Comment:
What is the purpose of this new property in state? If we need to track or compare the address to an edited value we can do that with a single property in state and the block attribute. The block attribute (`this.props.attributes.address`) should always hold the final and saved value. The state value should be used in the component. Only after clicking "apply" would the state value be saved as the block attribute value. And cancel should not update the block attribute (and we can reset the state value if we want).

Reply:
Thank you for pointing this out. I got a look today and I realized that I was clearly replicating the address state into two separate states. I went ahead and made the change to remove the addressText state. The block is tested and working as expected.